prover/src/lib.rs

```diff
@@ -323,19 +306,26 @@ pub trait Prover {
 
             trace_polys
                 .add_aux_segment(aux_segment_polys, air.context().lagrange_kernel_aux_column_idx());
-            aux_trace_rand_elements.set_segment_elements(rand_elements);
-            aux_trace_segment = Some(aux_segment);
-        }
+
+            Some(aux_trace_with_metadata)
+        } else {
+            None
+        };
+
+        let (aux_trace, aux_rand_elements, aux_proof) = match aux_trace_with_metadata {
+            Some(atm) => (Some(atm.aux_trace), Some(atm.aux_rand_eles), atm.aux_proof),
+            None => (None, None, None),
+        };
```

Comment:
nit: should this be merged with the previous statement block?

Reply:
Left it separate as a consequence on `trace.validate()` now taking `aux_trace_with_metadata` ([comment](https://github.com/facebook/winterfell/pull/271#discussion_r1586754298))